playwright/_impl/_fetch.py

```diff
@@ -0,0 +1,405 @@
+            headers=headers,
+            timeout=timeout,
+            failOnStatusCode=failOnStatusCode,
+            ignoreHTTPSErrors=ignoreHTTPSErrors,
+        )
+
+    async def patch(
+        self,
+        url: str,
+        params: Dict[str, Union[bool, float, str]] = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
+        multipart: Dict[str, Union[bytes, bool, float, str, FilePayload]] = None,
+        timeout: float = None,
+        failOnStatusCode: bool = None,
+        ignoreHTTPSErrors: bool = None,
+    ) -> "APIResponse":
+        return await self.fetch(
+            url,
+            method="PATCH",
+            params=params,
+            headers=headers,
+            data=data,
+            form=form,
+            multipart=multipart,
+            timeout=timeout,
+            failOnStatusCode=failOnStatusCode,
+            ignoreHTTPSErrors=ignoreHTTPSErrors,
+        )
+
+    async def put(
+        self,
+        url: str,
+        params: Dict[str, Union[bool, float, str]] = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
+        multipart: Dict[str, Union[bytes, bool, float, str, FilePayload]] = None,
+        timeout: float = None,
+        failOnStatusCode: bool = None,
+        ignoreHTTPSErrors: bool = None,
+    ) -> "APIResponse":
+        return await self.fetch(
+            url,
+            method="PUT",
+            params=params,
+            headers=headers,
+            data=data,
+            form=form,
+            multipart=multipart,
+            timeout=timeout,
+            failOnStatusCode=failOnStatusCode,
+            ignoreHTTPSErrors=ignoreHTTPSErrors,
+        )
+
+    async def post(
+        self,
+        url: str,
+        params: Dict[str, Union[bool, float, str]] = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
+        multipart: Dict[str, Union[bytes, bool, float, str, FilePayload]] = None,
+        timeout: float = None,
+        failOnStatusCode: bool = None,
+        ignoreHTTPSErrors: bool = None,
+    ) -> "APIResponse":
+        return await self.fetch(
+            url,
+            method="POST",
+            params=params,
+            headers=headers,
+            data=data,
+            form=form,
+            multipart=multipart,
+            timeout=timeout,
+            failOnStatusCode=failOnStatusCode,
+            ignoreHTTPSErrors=ignoreHTTPSErrors,
+        )
+
+    async def fetch(
+        self,
+        urlOrRequest: Union[str, network.Request],
+        params: Dict[str, Union[bool, float, str]] = None,
+        method: str = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
```

Comment:
Let's ask Yury about repeated keys.

Reply:
answered in https://github.com/microsoft/playwright-python/pull/1013#discussion_r745931594